bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -384,13 +384,13 @@ def extensions_dir
       end
     end
 
-    remove_method :ignored? if new.respond_to?(:ignored?)
+    # Can be removed once RubyGems 3.5.22 support is dropped
+    unless new.respond_to?(:ignored?)
```

Comment:
Although unrelated to this issue, why not `method_defined?(:ignored?)` without creating a new instance?

Reply:
No particular reason, just not explicitly concerned with performance given it should only run once. But happy to switch to `method_defined?`, can you send a PR?